homeassistant/components/mcp_server/http.py

```diff
@@ -0,0 +1,170 @@
+from mcp import types
+
+from homeassistant.components import conversation
+from homeassistant.components.http import KEY_HASS, HomeAssistantView
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_LLM_HASS_API
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import llm
+
+from .const import DOMAIN
+from .server import create_server
+from .session import Session
+from .types import MCPServerConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+SSE_API = f"/{DOMAIN}/sse"
+MESSAGES_API = f"/{DOMAIN}/messages/{{session_id}}"
+
+
+@callback
+def async_register(hass: HomeAssistant) -> None:
+    """Register the websocket API."""
+    hass.http.register_view(ModelContextProtocolSSEView())
+    hass.http.register_view(ModelContextProtocolMessagesView())
+
+
+def async_get_config_entry(hass: HomeAssistant) -> MCPServerConfigEntry:
+    """Get the first enabled MCP server config entry.
+
+    The ConfigEntry contains a reference to the actual MCP server used to
+    serve the Model Context Protocol.
+
+    Will raise an HTTP error if the expected configuration is not present.
+    """
+    config_entries: list[MCPServerConfigEntry] = [
+        config_entry
+        for config_entry in hass.config_entries.async_entries(DOMAIN)
+        if config_entry.state == ConfigEntryState.LOADED
+    ]
+    if not config_entries:
+        raise HTTPNotFound(body="Model Context Protocol server is not configured")
+    if len(config_entries) > 1:
+        raise HTTPNotFound(body="Found multiple Model Context Protocol configurations")
+    return config_entries[0]
+
+
+class ModelContextProtocolSSEView(HomeAssistantView):
+    """Model Context Protocol SSE endpoint."""
+
+    name = f"{DOMAIN}:sse"
+    url = SSE_API
+
+    async def get(self, request: web.Request) -> web.StreamResponse:
+        """Process SSE messages for the Model Context Protocol.
+
+        This is a long running request for the lifetime of the client session
+        and is the primary transport layer between the client and server.
+
+        Pairs of buffered streams act as a bridge between the transport protocol
+        (SSE over HTTP views) and the Model Context Protocol. The MCP SDK
+        manages all protocol details and invokes commands on our MCP server.
+        """
+        hass = request.app[KEY_HASS]
+        entry = async_get_config_entry(hass)
+        session_manager = entry.runtime_data
+
+        context = llm.LLMContext(
+            platform=DOMAIN,
+            context=self.context(request),
+            user_prompt=None,
+            language="*",
+            assistant=conversation.DOMAIN,
+            device_id=None,
+        )
+        llm_api_id = entry.data[CONF_LLM_HASS_API]
+        server = await create_server(hass, llm_api_id, context)
+        options = await hass.async_add_executor_job(
+            server.create_initialization_options
```

Comment:
I/O ?

Reply:
Yes, added a comment. This reads package ffnromation for getting the version.